### PR TITLE
[WIP] RavenDB-16705

### PR DESCRIPTION
**Pull request by ppekrol**
- [x] fixed NRe in BlittableJsonReaderObject.AssertNoModifications
- [x] fixed issue with replication a document tombstone with DeleteRevision flag
- [ ] write tests
- [ ] apply to 4.2

### CODE REVIEW

File: src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs [hunk 1]
```diff
@@ -373,12 +374,13 @@ private bool CanContinueBatch(ReplicationState state, ref long next)
         {
             if (MissingAttachmentsInLastBatch)
             {
-                state.MissingTxMarkers.Remove(state.LastTransactionMarker);
+                // we do have missing attachments but we haven't gathered yet any of the missing hashes
+                if (state.MissingAttachmentBase64Hashes == null)
```
Comment (ayende) [inline src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs:378]: I don't understand how that scenario can be?

File: test/SlowTests/Issues/EncryptedDatabaseGroup.cs [hunk 2]
```diff
@@ -28,12 +28,14 @@ public async Task AddingNodeToEncryptedDatabaseGroupShouldThrow()
             var options = new Options
             {
                 Server = leader,
-                ReplicationFactor = 2,
+                ReplicationFactor = 1,
```
Comment (ayende) [inline test/SlowTests/Issues/EncryptedDatabaseGroup.cs:31]: was this intentional?

Comment (ppekrol) [reply test/SlowTests/Issues/EncryptedDatabaseGroup.cs:31]: That's for debugging, all of those fixes need to be applied to 4.2 anyway.

File: src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs:378]: We are iterating items by etag, documents can be later on, but I do not want to break the batch before that? I think it should not be a case BTW, but prefer to stay on the safe side, I would like to not break the batch to early.

Comment (ayende) [reply src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs:378]: attachments will _always_ happen before their document.

**Pull request closed without merging**